src/main/scala/io/iohk/ethereum/network/AuthHandshaker.scala

```diff
@@ -192,4 +159,51 @@ case class AuthHandshaker(
     successOpt getOrElse AuthHandshakeError
   }
 
+  private def getRemoteEphemeralKey(initiateMessage: AuthInitiateMessage, responseMessage: AuthResponseMessage) = {
+    if (isInitiator) responseMessage.ephemeralPublicKey
+    else {
+      val agreement = new ECDHBasicAgreement
+      agreement.init(nodeKey.getPrivate)
+      val sharedSecret = agreement.calculateAgreement(new ECPublicKeyParameters(initiateMessage.publicKey, curve))
+
+      val token = bigIntegerToBytes(sharedSecret, NonceSize)
+      val signed = xor(token, initiateMessage.nonce.toArray)
+
+      val signaturePubBytes =
+        ECDSASignature.recoverPubBytes(
+          r = initiateMessage.signature.r,
+          s = initiateMessage.signature.s,
+          recId = ECDSASignature.recIdFromSignatureV(initiateMessage.signature.v),
+          message = signed).get
+
+      curve.getCurve.decodePoint(signaturePubBytes)
+    }
+  }
+
+  private def macSecretSetup(agreedSecret: Array[Byte],
+                             aesSecret: Array[Byte],
+                             initiatePacket: ByteString,
+                             initiateMessage: AuthInitiateMessage,
+                             responsePacket: ByteString,
+                             responseMessage: AuthResponseMessage) = {
+    val macSecret = sha3(agreedSecret, aesSecret)
+
+    val mac1 = new KeccakDigest(MacSize)
+    mac1.update(xor(macSecret, responseMessage.nonce.toArray), 0, macSecret.length)
+    val bufSize = 32
+    val buf = new Array[Byte](bufSize)
+    new KeccakDigest(mac1).doFinal(buf, 0)
+    mac1.update(initiatePacket.toArray, 0, initiatePacket.toArray.length)
+    new KeccakDigest(mac1).doFinal(buf, 0)
+
+    val mac2 = new KeccakDigest(MacSize)
+    mac2.update(xor(macSecret, initiateMessage.nonce.toArray), 0, macSecret.length)
+    new KeccakDigest(mac2).doFinal(buf, 0)
+    mac2.update(responsePacket.toArray, 0, responsePacket.toArray.length)
+    new KeccakDigest(mac2).doFinal(buf, 0)
+
+    if (isInitiator) (mac1, mac2)
+    else (mac2, mac1)
+  }
```

Comment:
It is as we currently we are currently using a rule that throws warnings if there is any method that has more than 50 lines. I think it is up to discussion whether or not we should use that rule as there are some cases where its not possible to reduce the number of lines of a method.